src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -28,6 +33,7 @@ public UrlHelper(ActionContext actionContext)
             }
 
             ActionContext = actionContext;
+            _stringBuilder = new StringBuilder();
```

Comment:
It would be valuable in some scenarios to make this field lazy. There are uses of the `UrlHelper` that don't call `GenerateUrl`